rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,311 @@
+                        notes=f'Spend {bonder_fee} {event.asset.symbol_or_name()} as a hop fee',
+                        counterparty=CPT_HOP,
+                        address=event.address,
+                    )
+                    context.decoded_events.append(fee_event)
+                    maybe_reshuffle_events(
+                        ordered_events=[event, fee_event],
+                        events_list=context.decoded_events,
+                    )
+                event.notes = self._get_bridge_note(
+                    event.balance.amount,
+                    event.asset,
+                    recipient,
+                    string_to_evm_address(event.location_label) if event.location_label else None,
+                    hex_or_bytes_to_int(context.tx_log.topics[2]),
+                )
+                break
+
+            if (
+                # send method (0xa6bd1b33) works differently than normal bridge events
+                # this is returning the hTOKEN to redeem the TOKEN in L1
+                # that's why we are considering it as a Burn below in the notes.
+                event.address == ZERO_ADDRESS and
+                event.asset.identifier == bridge.hop_identifier and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_HOP
+                event.notes = f'Burn {event.balance.amount} of Hop {event.asset.symbol_or_name()}'
+                break
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdrawal(self, context: DecoderContext) -> DecodingOutput:
+        """This function is used to decode the Withdrew event on Hop protocol."""
+        if not self.base.is_tracked(hex_or_bytes_to_address(context.tx_log.topics[2])):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (bridge := self.bridges.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = self._get_bridgeasset_amount(amount_raw, bridge)
+        for event in context.decoded_events:
+            if (
+                event.address in (bridge.amm_wrapper, bridge.address) and
+                event.asset.identifier == bridge.identifier and
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == amount
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.notes = self._get_bridge_note(event.balance.amount, event.asset)
+                event.counterparty = CPT_HOP
+                break
+
+        else:
```

Comment:
add a comment here on what this else is
```suggestion
        else:  # corresponding receive not found DOING BLA BLA BLA you know what you are doing better than I do
```